website/fiddle/public/auth/idpmock/logout.html

```diff
@@ -9,10 +9,19 @@
 	Logging out...
 	<script>
 		window.onload = function() {
-    const redirectTo = decodeURIComponent(
+    let redirectTo = decodeURIComponent(
       window.location.href.match(/post_logout_redirect_uri=(.*?)(&|$)/)[1]
     );
-    window.location.href = redirectTo;
+    const disableJs = redirectTo.replace(/javascript:/gi, '');

```

Comment:
I am not sure if this is enough, since there are tons of possibilities how one could write `javascript` according to https://owasp.org/www-community/xss-filter-evasion-cheatsheet  .. But since we are checking if it starts with _window.location.origin_ I couldn't find a possibility to sneak anything in. Just thinking if we remove that check at all just not to raise wrong safety feelings.

Reply:
I would say it's fine.
I tested:
`http://localhost:8080/auth/idpmock/logout.html?post_logout_redirect_uri=javascript:alert(&quot;XSS&quot;)
http://localhost:8080/auth/idpmock/logout.html?post_logout_redirect_uri=JaVaScRiPt:alert('XSS')
http://localhost:8080/auth/idpmock/logout.html?post_logout_redirect_uri=javascript:alert('XSS');
http://localhost:8080/auth/idpmock/logout.html?post_logout_redirect_uri=javascript:alert("RSnake says, 'XSS'")`
All of them aren't working.

We can think if we apply the last point in the issue description and replace `window.location.origin` with the fiddle url but I think we can also do the implemented version.